snc.sh

```diff
@@ -8,12 +8,19 @@ JQ=${JQ:-jq}
 OC=${OC:-oc}
 YQ=${YQ:-yq}
 OPENSHIFT_INSTALL=${OPENSHIFT_INSTALL:-./openshift-install}
-OPENSHIFT_RELEASE_VERSION=$(git describe --exact-match --tags HEAD 2>/dev/null) || OPENSHIFT_RELEASE_VERSION=
 CRC_VM_NAME=${CRC_VM_NAME:-crc}
 BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
 CRC_PV_DIR="/mnt/pv-data"
 SSH="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i id_rsa_crc"
 
+# If user defined the OPENSHIFT_VERSION environment variable then use it.
+# Otherwise use the tagged version if available
+function get_openshift_version {
+    OPENSHIFT_RELEASE_VERSION=$(git describe --exact-match --tags HEAD 2>/dev/null)
+    if [ ${OPENSHIFT_VERSION} != "" ]; then
+        OPENSHIFT_RELEASE_VERSION=$OPENSHIFT_VERSION
+    fi
```

Comment:
No need for the "git describe" when OPENSHIFT_VERSION is set? I believe this could be written as
```
if [ -z ${OPENSHIFT_VERSION} ]; then 
    OPENSHIFT_RELEASE_VERSION=$(git describe ...)
else 
    OPENSHIFT_RELEASE_VERSION=$OPENSHIFT_VERSION
fi
```
Maybe we could use "OPENSHIFT_RELEASE_VERSION" as the name of the environment variable rather than introducing an extra "OPENSHIFT_VERSION"? (just a thought, I'm fine either way)

Reply:
If a user set `OPENSHIFT_VERSION` then we are going to use that instead the tag, otherwise if the master branch have tagged then there is no way to deploy a different version then tagged one until switch to different branch or go to a different commit. So here we will give user the preference how they want to build.

Reply:
If current git master tip is tagged, we can now build a different openshift version (say 4.1.11 instead of 4.1.13), but we cannot build an unspecified openshift version (ie use whatever version the installer was built to use). Is this enough for what you want to do?

Reply:
Yes, I was not suggesting you change the behaviour of what you describe, I was just proposing alternative ways of doing the same, and avoid the "git describe" call when not needed.